integration/test/metrics_number_dimension/metrics_number_dimension_test.go

```diff
@@ -56,17 +54,26 @@ func TestNumberMetricDimension(t *testing.T) {
 			numberDimensionsInCW: 10,
 			metricName:           "mem_used_percent",
 		},
-		// @TODO add when https://github.com/aws/amazon-cloudwatch-agent/pull/361 is merged
-		// {resourcePath: "resources/30_dimension", findTarget: false, numberDimensionsInCW: 30, metricName: "mem_used_percent",},
-		// {resourcePath: "resources/35_dimension", findTarget: true, numberDimensionsInCW: 30, metricName: "mem_used_percent",},
+		{
+			resourcePath:         "resources/30_dimension",
+			findTarget:           false,
+			numberDimensionsInCW: 30,
+			metricName:           "mem_used_percent",
+		},
+		{
+			resourcePath:         "resources/35_dimension",
+			findTarget:           true,
+			numberDimensionsInCW: 30,
```

Comment:
Should this be 35?

Reply:
@SaxyPandaBear It should be 30 since CWAgent would drop metric's dimension if the dimensions > maxDimension based on my understanding.

Reply:
Yeah that is how it works for statsd. For append dimension it should fail on the config check. My bad on that.

Reply:
I rewrote the test for this.